### PR TITLE
Arodriguez/feature 148 hazard report form UI hazard category2

### PR DESCRIPTION
**Pull request by ALFAROSO**
Hazard Report Form UI - Hazard Category Mobile Layout

### CODE REVIEW

File: pages/report-hazard/index.html [hunk 1]
```diff
@@ -105,7 +105,8 @@
           </li>
         </ul>
       </nav>
-      <button id="backButton" class="back-button-mobile">&lt; Back</button>
+      <button id="backButton" class="back-button-mobile">Back</button>
+      <div class="underline"></div>
```
Comment (whamester) [inline pages/report-hazard/index.html:109]: can we rename this `underline` to `divider`?

**Comment by whamester** [comment]
Let's add a max height to the container so we don't have to scroll down to find the button:

Current:

https://github.com/whamester/WMDD-4885-Project/assets/57273227/447b3485-81f6-402b-b068-c6d23f7510a6



Expected:
https://github.com/whamester/WMDD-4885-Project/assets/57273227/dd05ae23-6f3c-40ac-8543-7bca84ee14af

I achieve this example by adding the following styles:

#hazard-category-content {
    max-height: calc(100vh - 300px);
    overflow: auto;
}

You can adjust if needed. This is just an example.

File: pages/report-hazard/script.js [hunk 2]
```diff
@@ -254,6 +254,14 @@ const getAddressFromCoordinates = async (params) => {
   }
 };
 
+//Insert chevron-left arrow on back button
+const button = document.getElementById('backButton');
+const icon = document.createElement('img');
```
Comment (whamester) [inline pages/report-hazard/script.js:259]: Did you try adding the icon with <i class="icon-chevron-left"></i>?

File: pages/report-hazard/script.js [hunk 3]
```diff
@@ -275,9 +283,25 @@ const getCategories = async () => {
...
+    data.forEach((category) => {
+      if (category.name && category.ui_settings.icon) {
+        const iconParts = category.ui_settings.icon.toLowerCase().split('-');
+        if (iconParts.length >= 2) {
+          arrayIcons.push(iconParts[1]);
+        }
+      }
+    });
+
+    console.log(arrayIcons);
```
Comment (whamester) [inline pages/report-hazard/script.js:297]: remove this console log

File: pages/report-hazard/script.js [hunk 4]
```diff
@@ -311,12 +335,28 @@ const getCategories = async () => {
 
       label.setAttribute('id', `category-${category.id}-label`);
       label.setAttribute('for', `category-${category.id}-radio`);
-      label.innerHTML = category.name;
+      label.classList.add('label-container');
+      label.innerHTML = `<i class="category-icon"><img src="../../assets/icons/${arrayIcons[i]}-outline.svg" alt="${arrayIcons[i]}"></i>`;
```
Comment (whamester) [inline pages/report-hazard/script.js:339]: you could use only the i tag here

<i class="${category.ui_settings.icon} category-icon"></i>

File: pages/report-hazard/style.css [hunk 5]
```diff
@@ -75,15 +73,128 @@ img {
...
+}
+
+/* Back button */
+#backButton {
+  border: none;
+  background: none;
+  color: inherit;
+  cursor: pointer;
+  text-decoration: none;
+  color: var(--neutral-neutral-500, #667185);
```
Comment (whamester) [inline pages/report-hazard/style.css:92]: this should be

color: var(--neutral-500);

File: pages/report-hazard/style.css [hunk 6]
```diff
@@ -75,15 +73,128 @@ img {
...
+  margin-bottom: 12px;
+  margin-left: 16px;
+  margin-right: 16px;
+  display: flex;
+  align-items: center;
+}
+
+.underline {
+  width: 100%;
+  border-bottom: 1px solid var(--neutral-neutral-300, #d0d5dd);
```
Comment (whamester) [inline pages/report-hazard/style.css:103]: same

File: pages/report-hazard/style.css [hunk 7]
```diff
@@ -75,15 +73,128 @@ img {
...
+}
+
+#hazard-category label,
+#hazard-option-content label {
+  padding: 5px 10px;
+  cursor: pointer;
+  display: inline-block;
+  margin: 5px;
+  border-radius: 8px;
+  border: 1px solid var(--neutral-neutral-300, #d0d5dd);
```
Comment (whamester) [inline pages/report-hazard/style.css:125]: same

File: pages/report-hazard/style.css [hunk 8]
```diff
@@ -75,15 +73,128 @@ img {
...
+  background: var(--neutral-neutral-50, #f7f9fc);
+}
+
+#hazard-category input[type='radio']:checked + label,
+#hazard-option-content input[type='radio']:checked + label {
+  border-radius: 8px;
+  border: 1px solid var(--neutral-neutral-100, #f0f2f5);
+  background: var(--secondary-secondary-500-base, #d8e7e2);
```
Comment (whamester) [inline pages/report-hazard/style.css:133]: same

File: pages/report-hazard/style.css [hunk 9]
```diff
@@ -75,15 +73,128 @@ img {
...
+}
+
+.label-container {
+  padding: 20px;
+  width: 342px;
+  height: 104px;
+}
+
+.title-hazard {
+  color: var(--primary-primary-500-base, #243832);
```
Comment (whamester) [inline pages/report-hazard/style.css:170]: same

File: pages/report-hazard/style.css [hunk 10]
```diff
@@ -75,15 +73,128 @@ img {
...
+  /* Small/Body-1/Bold (700) */
```
Comment (whamester) [inline pages/report-hazard/style.css:171]:
```
  /* Small/Body-1/Bold (700) */
  /* font-family: Instrument Sans; */
  font-size: 16px;
  font-style: normal;
  font-weight: 700;
  line-height: 150%; /* 24px */
```

These are not needed we have global classes for the text, reach out and we can review this together.

File: pages/report-hazard/style.css [hunk 11]
```diff
@@ -75,15 +73,128 @@ img {
...
+  font-style: normal;
+  font-weight: 700;
+  line-height: 150%; /* 24px */
+  margin-bottom: 4px;
+}
+
+.description-hazard {
+  color: var(--primary-primary-400, #50605b);
+
+  /* Small/Body-3/Medium (500) */
```
Comment (whamester) [inline pages/report-hazard/style.css:183]: Same with these


```
  /* Small/Body-1/Bold (700) */
  /* font-family: Instrument Sans; */
  font-size: 16px;
  font-style: normal;
  font-weight: 700;
  line-height: 150%; /* 24px */

```

File: pages/report-hazard/style.css [hunk 12]
```diff
@@ -75,15 +73,128 @@ img {
...
+  /* font-family: Instrument Sans; */
+  font-size: 16px;
+  font-style: normal;
+  font-weight: 700;
+  line-height: 150%; /* 24px */
+  margin-bottom: 4px;
+}
+
+.description-hazard {
+  color: var(--primary-primary-400, #50605b);
```
Comment (whamester) [inline pages/report-hazard/style.css:181]: same

File: pages/report-hazard/script.js (hunk 4 above)
Comment (ALFAROSO) [reply pages/report-hazard/script.js:339]: Do you mean `<i>` tag or `` tag?
`label.innerHTML = "${arrayIcons[i]}";`

File: pages/report-hazard/style.css (hunk 5 above)
Comment (ALFAROSO) [reply pages/report-hazard/style.css:92]: Done

File: pages/report-hazard/style.css (hunk 6 above)
Comment (ALFAROSO) [reply pages/report-hazard/style.css:103]: Done

File: pages/report-hazard/style.css (hunk 7 above)
Comment (ALFAROSO) [reply pages/report-hazard/style.css:125]: Done

File: pages/report-hazard/style.css (hunk 12 above)
Comment (ALFAROSO) [reply pages/report-hazard/style.css:181]: done

File: pages/report-hazard/style.css (hunk 9 above)
Comment (ALFAROSO) [reply pages/report-hazard/style.css:170]: Done

File: pages/report-hazard/style.css (hunk 8 above)
Comment (ALFAROSO) [reply pages/report-hazard/style.css:133]: Done

File: pages/report-hazard/script.js (hunk 3 above)
Comment (ALFAROSO) [reply pages/report-hazard/script.js:297]: Done

File: pages/report-hazard/style.css (hunk 11 above)
Comment (ALFAROSO) [reply pages/report-hazard/style.css:183]: done

File: pages/report-hazard/style.css (hunk 10 above)
Comment (ALFAROSO) [reply pages/report-hazard/style.css:171]: done

File: pages/report-hazard/index.html (hunk 1 above)
Comment (ALFAROSO) [reply pages/report-hazard/index.html:109]: done

**Review by whamester — APPROVED** [review]

**Pull request merged**